internal/controllers/infobloxippool.go

```diff
@@ -137,6 +137,19 @@ func (r *InfobloxIPPoolReconciler) reconcile(ctx context.Context, pool *v1alpha1
 		pool.Spec.NetworkView = ibclient.GetHostConfig().DefaultNetworkView
 	}
 
+	if pool.Spec.DNSView == "" {
+		if ibclient.GetHostConfig().DefaultDNSView != "" {
+			pool.Spec.DNSView = ibclient.GetHostConfig().DefaultDNSView
```

Comment:
Not sure if we should set it in the spec. The Patch Helper will write it to the API server, and changing the zone for all pools referencing an instance would then require updating the value everywhere. I think we should not set it in the spec and just calculate it every time instead.

It would also be nice to move this logic into a method/function that can be called, e.g. `determineDNSZone()`. The resulting zone can then be kept as a variable during reconciliation.